File: src/main/typescript/wcardinal/controller/internal/info/rejected-set-dynamic-info.ts
/*
 * Copyright (C) 2019 Toshiba Corporation
 * SPDX-License-Identifier: Apache-2.0
 */

import { now } from "../../../util/lang/now";
import { DynamicInfo } from "../settings";
import { RejectionInfo } from "./rejection-info";

export class RejectedDynamicInfo {
	public readonly _rejectionInfo: RejectionInfo;
	public readonly _dynamicInfo: DynamicInfo;
	public readonly _createdAt: number;

	constructor( rejectionInfo: RejectionInfo, dynamicInfo: DynamicInfo ) {
		this._rejectionInfo = rejectionInfo;
		this._dynamicInfo = dynamicInfo;
		this._createdAt = now();
	}
}

File: src/main/java/org/wcardinal/controller/internal/info/RejectedSetDynamicInfo.java
/*
 * Copyright (C) 2019 Toshiba Corporation
 * SPDX-License-Identifier: Apache-2.0
 */

package org.wcardinal.controller.internal.info;

public class RejectedSetDynamicInfo {
	public final SetDynamicInfo info;
	public final ReceivedDynamicInfo receivedInfo;
	public final long createdAt;

	public RejectedSetDynamicInfo( final SetDynamicInfo info, final ReceivedDynamicInfo receivedInfo ) {
		this.info = info;
		this.receivedInfo = receivedInfo;
		this.createdAt = System.currentTimeMillis();
	}
}
